src/EFCore.PG/Query/Internal/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -470,6 +470,55 @@ protected override Expression VisitNew(NewExpression newExpression)
                 : QueryCompilationContext.NotTranslatedExpression;
         }
 
+        // Translate new NpgsqlCube(...) -> cube(...)
+        if (newExpression.Constructor?.DeclaringType == typeof(NpgsqlCube))
+        {
+            if (!TryTranslateArguments(out var sqlArguments))
+            {
+                return QueryCompilationContext.NotTranslatedExpression;
+            }
+
+            var cubeTypeMapping = _typeMappingSource.FindMapping(typeof(NpgsqlCube));
+
+            // Distinguish constructor overloads by argument count and types
+            switch (sqlArguments.Length)
+            {
+                case 1:
+                    // NpgsqlCube(double coord) or NpgsqlCube(IEnumerable<double> coords)
+                    // Both map to cube() - PostgreSQL overloads based on argument type
+                    return _sqlExpressionFactory.Function(
+                        "cube",
+                        sqlArguments,
+                        nullable: false,
+                        argumentsPropagateNullability: FalseArrays[1],
```

Comment:
FalseArrays here means that if you pass NULL to the `cube()` PG function, it returns non-NULL, which I doubt - almost all SQL functions propagate nullability (i.e. they return null if one of their arguments is null). Can you check this, and if needed, correct the argument here?